ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -16,174 +15,226 @@ import com.daml.lf.transaction.GlobalKey
-      ) extends LoggingTransactionErrorImpl(
-            cause = message
-          )
+    object Preprocessing extends ErrorGroup {
+      @Explanation("""This error occurs if a command fails during interpreter pre-processing.""")
+      @Resolution("Inspect error details and correct your application.")
+      object PreprocessingFailed
+          extends ErrorCode(
+            id = "COMMAND_PREPROCESSING_FAILED",
+            ErrorCategory.InvalidIndependentOfSystemState,
+          ) {
+        case class Reject(
+            err: LfError.Preprocessing.Error
+        )(implicit
+            loggingContext: ContextualizedErrorLogger
+        ) extends LoggingTransactionErrorImpl(
+              cause = err.message
+            )
+      }
     }
-  }
 
-  object ReadErrors extends ErrorGroup() {
+    object Interpreter extends ErrorGroup {
+      @Explanation("""This error occurs if a Daml transaction fails during interpretation.""")
+      @Resolution("This error type occurs if there is an application error.")
+      object GenericInterpretationError
+          extends ErrorCode(
+            id = "DAML_INTERPRETATION_ERROR",
+            ErrorCategory.InvalidGivenCurrentSystemStateOther,
+          ) {
 
-    @Explanation("This rejection is given when a package id is malformed.")
-    @Resolution("Make sure the package id provided in the request has correct form.")
-    // TODO error codes: Consider using `LedgerApiErrors.CommandValidation.InvalidArgument`
-    object MalformedPackageId
-        extends ErrorCode(
-          id = "MALFORMED_PACKAGE_ID",
-          ErrorCategory.InvalidIndependentOfSystemState,
-        ) {
-      case class Reject(message: String)(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = message
-          )
-    }
+        case class Error(override val cause: String)(implicit
+            loggingContext: ContextualizedErrorLogger
+        ) extends LoggingTransactionErrorImpl(
+              cause = cause
+            )
+      }
 
-    @Explanation(
-      "This rejection is given when a read request tries to access a package which does not exist on the ledger."
-    )
-    @Resolution("Use a package id pertaining to a package existing on the ledger.")
-    // TODO error codes: Possible duplicate of `LedgerApiErrors.Package.MissingPackage`
-    object PackageNotFound
-        extends ErrorCode(
-          id = "PACKAGE_NOT_FOUND",
-          ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
-        ) {
-      case class Reject(packageId: String)(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = "Could not find package."
+      @Explanation(
+        """This error occurs if a Daml transaction fails during interpretation due to an invalid argument."""
+      )
+      @Resolution("This error type occurs if there is an application error.")
```

Comment:
This does not tell the user how to resolve the issue. An alternative could be `Ensure that your application generates a valid Daml transaction.`.